internal/api/v2/handler_payments_create.go

```diff
@@ -0,0 +1,226 @@
+package v2
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"net/http"
+	"time"
+
+	"github.com/formancehq/go-libs/api"
+	"github.com/formancehq/go-libs/pointer"
+	"github.com/formancehq/payments/internal/api/backend"
+	"github.com/formancehq/payments/internal/models"
+	"github.com/formancehq/payments/internal/otel"
+)
+
+type createPaymentRequest struct {
+	Reference            string            `json:"reference"`
+	ConnectorID          string            `json:"connectorID"`
+	CreatedAt            time.Time         `json:"createdAt"`
+	Type                 string            `json:"type"`
+	Amount               *big.Int          `json:"amount"`
+	Asset                string            `json:"asset"`
+	Scheme               string            `json:"scheme"`
+	Status               string            `json:"status"`
+	SourceAccountID      *string           `json:"sourceAccountID"`
+	DestinationAccountID *string           `json:"destinationAccountID"`
+	Metadata             map[string]string `json:"metadata"`
+}
+
+func (r *createPaymentRequest) validate() error {
```

Comment:
In a future PR let's consider normalising the validation step using a library so we don't have to write custom validation functions like this for every endpoint